### PR TITLE
fix crosslink epoch refs to begin with GENESIS_EPOCH; more minor epoch processing tweaks/unstubbed-code

### PR DESCRIPTION
**Pull request by tersec**

### CODE REVIEW

File: beacon_chain/beacon_node.nim [hunk 1]
```diff
@@ -202,7 +202,7 @@ proc makeAttestation(node: BeaconNode,
       beacon_block_root: node.headBlockRoot,
       epoch_boundary_root: Eth2Digest(), # TODO
       shard_block_root: Eth2Digest(), # TODO
-      latest_crosslink: Crosslink(), # TODO
+      latest_crosslink: Crosslink(epoch: state.latest_crosslinks[shard].epoch), # TODO
```
Comment (arnetheduck) [inline beacon_chain/beacon_node.nim]: still todo?

File: beacon_chain/spec/beaconstate.nim [hunk 2]
```diff
@@ -398,7 +401,12 @@ proc checkAttestation*(
       Crosslink(
         shard_block_root: attestation.data.shard_block_root,
         epoch: slot_to_epoch(attestation.data.slot))]):
-    warn("Unexpected crosslink shard")
+    warn("Unexpected crosslink shard",
+      state_latest_crosslinks_attestation_data_shard =
+        state.latest_crosslinks[attestation.data.shard],
+      attestation_data_latest_crosslink = attestation.data.latest_crosslink,
+      epoch = slot_to_epoch(attestation.data.slot),
```
Comment (arnetheduck) [inline beacon_chain/spec/beaconstate.nim]: `humaneEpochNum`

File: beacon_chain/beacon_node.nim (hunk 1 above)
Comment (tersec) [reply beacon_chain/beacon_node.nim]: Guess not anymore. Removed said comment.

File: beacon_chain/spec/beaconstate.nim (hunk 2 above)
Comment (tersec) [reply beacon_chain/spec/beaconstate.nim]: Added.

**Pull request merged**